tests/test_receivers.py

```diff
@@ -165,3 +165,35 @@ def test_event_deletion(app, receiver):
         event.delete()
         assert event.response == {'status': 410, 'message': 'Gone.'}
         assert event.response_code == 410
+
+
+def test_event_status(app):
+    """Test event status reporting."""
+    data = json.dumps({'x': 40, 'y': 2})
+
+    class TestCeleryReceiver(CeleryReceiver):
+
+        def run(self, event):
+            """Change task state."""
+            from celery import current_task, states
+
+            event.response['message'] = event.payload['x'] + event.payload['y']
```

Comment:
Changing the `process_event` task session from `with db.session.begin_nested:` to `db.session.begin(nested=True, subtransactions=True)` seems to solve the problem.